runtime/criusupport/criusupport.c

```diff
@@ -121,7 +98,7 @@ Java_org_eclipse_openj9_criu_CRIUSupport_isCheckpointAllowed(JNIEnv *env, jclass
 }
 
 static BOOLEAN
-getNativeString(J9VMThread *currentThread, j9object_t javaString, char **nativeString, jfieldID *resultType)
+getNativeString(J9VMThread *currentThread, j9object_t javaString, char **nativeString, const char **nlsMsgFormat, IDATA *errCode)
```

Comment:
This is a very odd mix of native error handling and Java exceptions.   Step back and look at the purpose of this method - it's to convert a javaString to a native string.  Nothing in that purpose needs to setting Java exceptions.

It's much cleaner if this returns null on failure or provides an error code to differentiate between success, allocation failure and conversion failure.

Let the caller of this method `setNativeOutOfMemoryError` if it wants to throw an exception.  Its much harder to ensure the code isn't losing an object pointer across a gc point is to not introduce unnecessary gc points by throwing exceptions in inner helper methods.

Reply:
I'm also not clear why the exceptions need this level of detail.  I'd suggest adding tracepoints for special error cases and otherwise let the caller throw a generic exception if there isn't user relevant data here (and there doesn't appear to be)